src/world_interface/kinematic_common_interface.cpp

```diff
@@ -77,6 +84,100 @@ double setCmdVel(double dtheta, double dx) {
 	return maxAbsPWM > 1 ? maxAbsPWM : 1.0;
 }
 
+double setTankCmdVel(double left, double right) {
+	if (Globals::E_STOP && (left != 0 || right != 0)) {
+		return 0;
+	}
+
+	for (int i = 1; i <= 4; i++) {
+		if (std::abs(robot::getMotorPos(Constants::WHEEL_IDS[i]) -
+					 Constants::NORMAL_WHEEL_ROTS[i]) < Constants::STEER_EPSILON)
+			return 0;
+	}
+
+	wheelvel_t wheelVels = {left, right};
+	double lPWM = wheelVels.lVel / Constants::MAX_WHEEL_VEL;
+	double rPWM = wheelVels.rVel / Constants::MAX_WHEEL_VEL;
+	double maxAbsPWM = std::max(std::abs(lPWM), std::abs(rPWM));
+	if (maxAbsPWM > 1) {
+		lPWM /= maxAbsPWM;
+		rPWM /= maxAbsPWM;
+	}
+
+	setCmdVelToIntegrate(wheelVels);
+	setMotorPower(motorid_t::frontLeftWheel, lPWM);
+	setMotorPower(motorid_t::rearLeftWheel, lPWM);
+	setMotorPower(motorid_t::frontRightWheel, rPWM);
+	setMotorPower(motorid_t::rearRightWheel, rPWM);
+
+	return maxAbsPWM > 1 ? maxAbsPWM : 1.0;
+}
+
+double setTurnInPlaceCmdVel(double dtheta) {
+	if (Globals::E_STOP && (dtheta != 0)) {
+		return 0;
+	}
+
+	for (int i = 1; i <= 4; i++) {
+		if (std::abs(robot::getMotorPos(Constants::WHEEL_IDS[i]) -
+					 Constants::TURN_IN_PLACE_WHEEL_ROTS[i]) < Constants::STEER_EPSILON)
+			return 0;
+	}
+
+	swervewheelvel_t wheelVels = swerveKinematics().robotVelToWheelVel(0, 0, dtheta);
+	double lfPWM = wheelVels.lfVel / Constants::MAX_WHEEL_VEL;
+	double lbPWM = wheelVels.lbVel / Constants::MAX_WHEEL_VEL;
+	double rfPWM = wheelVels.rfVel / Constants::MAX_WHEEL_VEL;
+	double rbPWM = wheelVels.rbVel / Constants::MAX_WHEEL_VEL;
+	double maxAbsPWM = std::max(std::max(std::abs(lfPWM), std::abs(lbPWM)),
+								std::max(std::abs(rfPWM), std::abs(rbPWM)));
+	if (maxAbsPWM > 1) {
+		lfPWM /= maxAbsPWM;
+		lbPWM /= maxAbsPWM;
+		rfPWM /= maxAbsPWM;
+		rbPWM /= maxAbsPWM;
+	}
+
+	setMotorPower(motorid_t::frontLeftWheel, lfPWM);
+	setMotorPower(motorid_t::rearLeftWheel, lbPWM);
+	setMotorPower(motorid_t::frontRightWheel, rfPWM);
+	setMotorPower(motorid_t::rearRightWheel, rbPWM);
+
+	return maxAbsPWM > 1 ? maxAbsPWM : 1.0;
+}
+
+double setCrabCmdVel(double dtheta, double dy) {
```

Comment:
This method does not handle rotation correctly at the moment. I wasn't sure how to, because we want this to essentially act as the normal drive mode but with a rotated reference frame